src/engine/parsers/SquadronFactory.java

```diff
@@ -96,7 +97,13 @@ public SquadronFactory(String pathToFile, boolean renderSquadrons) throws FileNo
                             this.defenseTokens = new ArrayList<>();
                         }else{
                             String[] defenseTokenArray = value.split(" ");
-                            this.defenseTokens = new ArrayList<>(Arrays.asList(defenseTokenArray));
+                            ArrayList<DefenseToken> tokens = new ArrayList<>();
+
+                            for(String token: defenseTokenArray){
+                                ParsingUtils.checkValidDefenseToken(token);
+                                tokens.add(new DefenseToken(token));
+                            }
+                            this.defenseTokens = tokens;
                         }
```

Comment:
This is my inner battlecode complaining. If both our if and else blocks create a new arraylist<>(), why don't default ships just have a new 0 length arraylist right out the gate, and if value !equals none, add stuff per parsing utils